AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -2342,8 +2374,16 @@ - (ASLayout *)calculateLayoutThatFits:(ASSizeRange)constrainedSize
 
   ASDN::MutexLocker l(__instanceLock__);
   if ((_methodOverrides & ASDisplayNodeMethodOverrideLayoutSpecThatFits) || _layoutSpecBlock != NULL) {
-    ASLayoutSpec *layoutSpec = [self layoutSpecThatFits:constrainedSize];
-    
+    ASLayoutSpec *layoutSpec = nil;
+    // optional performance measurement for cell nodes
+    if (self.measurementOptions & ASDisplayNodePerformanceMeasurementOptionsLayoutSpec) {
```

Comment:
Don't need to use the setter here since you've already grabbed the instance lock